PYIC-8381: create new SIS stub

thebauSoftwire opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

- addition of new SIS stub

**Testing notes:**
| case | notes|
|-|-|
| 200 | Used an access token generated by the api tests (I let a successful journey complete so that the associated userId had an SI record created by core-back) |
| 200 - Malformed JSON | Created an SI record via AWS console with missing vot/storedIdentity for a given user. Used the api tests to create an access token with the same userId. |
| 400 | Created an access token via the method above but removed the userId, missing request body, missing body.vtr, missing body.govukSigninJourneyId |
| 401 | Made a request without an auth header and with an incorrect bearer type |
| 403 | Made a request with a dummy access token |
| 404 | Used an access token generated by the api tests but for a scenario which wouldn't create an SI record) |
| 500 | deployed an instance of the stub which always returned a 500 |

### Why did it change

As part of phase 2 for SIS, we need to verify the stored identity record with the user's current VC bundle. To do this, we need to pull the SI record using a SIS endpoint. In lower environments, we need to interact with a stub instead.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-8381](https://govukverify.atlassian.net/browse/PYIC-8381)

## Checklists

## Checklists

<!-- Delete if changes in READMEs or documentation are not required -->
- [x] All READMEs and documentation updated where necessary

<!-- Delete if changes don't include risk of credentials being exposed -->
- [x] No risk of PII, credentials or anything else sensitive being exposed through logs

<!-- Delete if changes don't apply -->
- [x] Tests have been written/updated


[PYIC-8381]: https://govukverify.atlassian.net/browse/PYIC-8381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

thebauSoftwire committed:
PYIC-8381: add cloudformation templates for sis stub and openAPI spec

thebauSoftwire committed:
PYIC-8381: initialise ts project, add .gitignore, eslint config

thebauSoftwire committed:
PYIC-8381: add handling for getUserIdentity

thebauSoftwire committed:
PYIC-8381: improve logging for malformed JSON

thebauSoftwire committed:
PYIC-8381: add unit tests

thebauSoftwire committed:
PYIC-8381: add github workflow for deploying to stubs accounts

thebauSoftwire committed:
PYIC-8381: add README for sis-stub

thebauSoftwire commented on .github/workflows/sis-stub.yml:
```diff
@@ -0,0 +1,66 @@
+      - name: Set up AWS creds
+        uses: aws-actions/configure-aws-credentials@v4.3.1
+        with:
+          role-to-assume: ${{ secrets.SIS_STUB_BUILD_GH_ROLE }}
```
not sure where to find these but this is also a reminder to set these before merging

thebauSoftwire replied:
GH secrets have been added for SIS_STUB_BUILD_GH_ROLE and SIS_STUB_BUILD_S3_BUCKET

thebauSoftwire committed:
PYIC-8381: add default values for new parameters in cloudformation so we don't have to define them anywhere else

blakeyp commented on di-ipv-sis-stub/lambdas/src/services/sisService.ts:
```diff
@@ -0,0 +1,55 @@
+      // defaulting to false as the ttl is set to the default
+      // retention of VCs which is 120 years
+      expired: false,
+    };
```
There are a couple of extra fields `kidValid` and `signatureValid` which we can just default to true (they're new additions so not covered in the ticket sorry)

see https://github.com/govuk-one-login/ipv-identity-reuse-service/blob/12bac6383cfe76b74f738cd4e6998cc86a9439f8/src/specs/api.yaml#L88

blakeyp commented on di-ipv-sis-stub/openAPI/sis-external.yaml:
```diff
@@ -0,0 +1,170 @@
+            responseTemplates:
+              application/json: "{\"healthcheck\": \"ok\"}"
+  /user-identity:
+    get:
```
oo also sorry it's `POST` now, since they'd like us to pass the journey id with the request - see https://github.com/govuk-one-login/ipv-identity-reuse-service/blob/12bac6383cfe76b74f738cd4e6998cc86a9439f8/src/specs/api.yaml#L59 for the shape of the request body

thebauSoftwire committed:
PYIC-8381: update endpoint based on schema updates
- update endpoint from GET to POST
- add new request body
- add new response attributes
- rename functions/handlers to match endpoint type
